packages/block-editor/src/components/use-style/index.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * External dependencies
+ */
+import { get } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { useBlockEditContext } from '../block-edit';
+import { store as blockEditorStore } from '../../store';
+import { getResolvedStyleVariable } from '../../utils/style-variable-resolution';
+
+/**
+ * Hook that retrieves the global styles of a block.
+ * It works with nested objects using by finding the value at path.
+ *
+ * @param {string|Array} path The path to the setting.
+ *
+ * @return {any} Returns the style value defined for the path.
+ *
+ * @example
+ * ```js
+ * const backgroundColor = useStyle( 'color.background' );
+ * ```
+ */
+export default function useStyle( path ) {
+	const { name: blockName } = useBlockEditContext();
+
+	const setting = useSelect(
```

Comment:
I'm getting ahead of myself here, so sorry if this is premature optimization, but I was just curious as to whether this would, in the future, be a good case to use `createSelector`

Reply:
We already have a useSetting for the theme.json features as a hook so I'm trying to make the styles part consistent with that. I included a useMemo on the variable resolution part so now we have the same advantage as using createSelector.